### PR TITLE
Scenarios Authoring Documentation main batch of changes

### PR DESCRIPTION
**Pull request by ZbyszekMM**
Contains all related Scenarios Authoring docs and clean up of the DesigningProcesses folder.

### CODE REVIEW

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+Let’s assume that all the above events qualify to the time windows discussed below.
+
+
+## Common parameters
+
+Majority of parameters are shared among all the three window types implemented in Nussknacker. Parameters which are unique to a particular window type are described in the section specific for particular window type further down on this page. 
+
+Parameters taken by the nodes used to configure aggregates in time windows are easiest explained by the analogy to the SQL statement with a GROUP BY clause and an aggregating function: 
+
+SELECT AGGREGATOR_FUNCTION(COLUMN_A)
```
Comment (arkadius) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:33]: Please wrap with "code block"

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+## Sliding-window
+
+In our implementation of the sliding window the aggregation computation is triggered only when an event enters the window. This means that whenever an avent arrives to the Sliding-window for evaluation, Nussknacker computes the aggregate taking into account all the *preceding* events which qualify into the sliding window. 
+
+Parameters specific to the Sliding window:
+
+**windowLength **- just length of the sliding window
+
+**emitWhenEventLeft **- the aggregate computation can be also triggered when an event leaves the window. This means that the aggregate is computed taking into account all the *subsequent* events which qualify into the sliding window.
```
Comment (arkadius) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:95]: Please update content as was discussed on docs

File: docs/scenarios_authoring/BasicNodes.md [hunk 3]
```diff
@@ -1,69 +1,65 @@
...
 
-![union_window](../img/union_window.png)
+![union_window](img/union_window.png)
 
 Union merges multiple branches into one stream. For each incoming branch two parameters are configured:
 - key - it's value should be of type `String`, definex how elements from branches will be matched together
 - value - this is the output value which will be put the field with name the same as branch id
 
-Union node defines new stream which is union of all branches. In this new stream there is only one variable, it's name
+Union node defines new stream which is union of both branches. In this new stream there is only one variable, it's name
```
Comment (arkadius) [inline docs/scenarios_authoring/BasicNodes.md:62]: Union can join many branches - not only two

File: docs/scenarios_authoring/Subprocesses.md [hunk 4]
```diff
@@ -1,22 +1,22 @@
-#Subprocesses
+# Subprocesses
```
Comment (dswiecki) [inline docs/scenarios_authoring/Subprocesses.md:1]: One nastly thing is that we already renamed subprocesses to fragments...
And that images you have put in have subprocess in labels

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+![alt_text](img/tumblingWindow.png "tumbling-window")
+
+It is important to note that when the tumbling window closes, a new event is generated. Its timestamp is equal to the time of the timer that generated it, not system time of the moment when it happened. In other words the timestammp of the newly generated event which contains the aggregate will continue to use the notion of time used by events which  Tumbling-window saw. As none of the events which entered the Tumbling-window will be passed downstream, the variable containing data of the events which entered Tumbling-window will not be availabe downstream. For example, if these were Kafka events which were an input to the Tumbling-Window, the #input variable will not be available downstream of the Tumbling-window node. 
+
+
+
+## Sliding-window
+
+In our implementation of the sliding window the aggregation computation is triggered only when an event enters the window. This means that whenever an avent arrives to the Sliding-window for evaluation, Nussknacker computes the aggregate taking into account all the *preceding* events which qualify into the sliding window.
```
Comment (mproch) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:89]: an event

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+
+* [Role of Schema Registry](/about/TypicalImplementation)
+* [Confluent Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html)
+* [Kafka topic key](https://kafka.apache.org/intro)
+* [Flink keyed state](https://ci.apache.org/projects/flink/flink-docs-master/docs/concepts/stateful-stream-processing/#keyed-state)
+
+
+### Concepts
+
+Kafka topics are native streaming data input to Nussknacker and the native output where results of Nussknacker scenarios processing are placed. In Nussknacker terminology input topics are called _sources_, output topics are called _sinks_. This section discusses Kafka based  source and sink components. There are a couple different types of source and sinks components available in Nussknacker, the main difference is in how the information on the data schema is provided to Nussnacker.
```
Comment (mproch) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:16]: Nussnacker = Nussknacker

File: docs/scenarios_authoring/DataSourcesAndSinks.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+All JSON data types are allowed ; `java.lang.String, Number` refer to Java classes responsible for handling the given data type. The `java.lang` prefix is optional. 
+
+Similarly to Kafka-avro source, Nussknacker Designer knows in this case what data to expect in the Kafka events: it will hint field names and will perform additional validations on SpEL expressions involving data from `audit` topic.
+
+The disadvantage of using Kafka-typed-json as a data source is that whenever the `audit` topic will be used in the Kafka-typed-json source, its schema will have to be defined. This works well if there is only one occurrence of an audit topic in all Nussknacker scenarios; it may become difficult to maintain otherwise. 
+
+
+### No type information + JSON serialization
+
+There is only one component in this group -  Kafka-json sourVRce. The content of the JSON formatted Kafka event does not need to be declared in Nussknacker Designer. The values of the JSON object can still be accessed using the `#input.field-name `notation, however no hints will be available and no validations related to field names and their data types will be performed.
```
Comment (mproch) [inline docs/scenarios_authoring/DataSourcesAndSinks.md:136]: sourVRce => source

File: docs/scenarios_authoring/AggregatesInTimeWindows.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+* Min - returns minimal value
+* Max - computes maximal value
+* Sum - computes sum of values
+* List - returns list of inputs received by the aggregator; see aggregateBy to understand what is meant by inputs
+* Set - the result is a set of inputs received by the aggregator. Can be very ineffective for large sets, try to use ApproximateSetCardinality in this case
+* ApproximateSetCardinality - computes approximate cardinality of set using [HyperLogLog](https://en.wikipedia.org/wiki/HyperLogLog) algorithm.
+
+**output** - name of the variable which will hold the result of the aggregator.
+
+**keyBy** - equivalent of the GROUP BY in SQL; a result of the aggregator will be computed for each distinct keyBy value found by Nussknacker in the time window.
```
Comment (dswiecki) [inline docs/scenarios_authoring/AggregatesInTimeWindows.md:58]: groupBy

**Review by dswiecki — APPROVED** [review]

**Pull request merged**